lxd/patches.go

```diff
@@ -65,3 +71,21 @@ func patchesApplyAll(d *Daemon) error {
 }
 
 // Patches begin here
+func patchInvalidProfileNames(name string, d *Daemon) error {
+	profiles, err := dbProfiles(d.db)
+	if err != nil {
+		return err
+	}
+
+	for _, profile := range profiles {
+		if strings.Contains(profile, "/") || shared.StringInSlice(profile, []string{".", ".."}) {
```

Comment:
Nope, that's intentional as "..." works, so does ".blah". Those are weird names but are perfectly encodable as a URL and so there's no real reason for us to reject them.